scripts/updateapilist.py

```diff
@@ -48,6 +84,26 @@ def __repr__(self):
         return repr((self.release_level, self.title))
 
 
+@dataclass
+class Extractor:
+    path_format: str
+    response_key: str
+    is_split_repo: bool = True
+
+    
+    def client_for_repo(self, repo_slug) -> Optional[CloudClient]:
+        path = self.path_format.format(repo_slug=repo_slug)
+        url = f"{RAW_CONTENT_BASE_URL}/{path}/{REPO_METADATA_FILENAME}"
+        response = requests.get(url)
+        if response.status_code != requests.codes.ok:
+            return
+
+        return CloudClient(response.json())
+    
+    def get_clients_from_batch_response(self, response_json) -> List[CloudClient]:
+        return [self.client_for_repo(repo[self.response_key]) for repo in response_json if (not self.is_split_repo or allowed_split_repo(repo))]
```

Comment:
If you have the `if` here, then it stops being a data class (a template for actions, in our case) and it is knowing how to do things differently depending on `is_split_repo`  in this line, but using provided data provided by decisions made outside the class in the other lines. I don't think we should mix-and-match the two patterns. I suggest either:

(a)  generalizing `allowed_split_repo` (see my comment in the review just preceding this one) and getting rid of `is_split_repo` here

or

(b) Making this a regular class with one parameter `is_split_repo`, where the constructor can set, according to that value, `self.path_format`, `self.response_key` and `self.do_include`. `self.do_include` can be assigned one of two functions: a new function always returning true (in the monorepo case) or  `allowed_split_repo` (in the split repo case). Then the list comprehension in this line looks like `[ ... if self.do_include(repo)]`. To be clear, what I have in mind for the constructor is something along the lines of 

```python
def __init__(self, is_split_repo):
  this.path_format = XXX if is_split_repo else YYY
  this.response_key = XXX if is_split_repo else YYY
  this.do_include = allowed_split_repo if is_split_repo else lambda x: True
```